nautobot_ssot/integrations/dna_center/diffsync/adapters/dna_center.py

```diff
@@ -72,10 +72,41 @@ def load_locations(self):
 
     def load_controller_locations(self):
         """Load location data for Controller specified in Job form."""
-        if self.job.dnac.location.location_type.name == "Site":
+        if self.job.dnac.location.location_type == self.job.floor_loctype:
+            self.get_or_instantiate(
+                self.floor,
+                ids={
+                    "name": self.job.dnac.location.name,
+                    "building": self.job.dnac.location.parent.name,
+                    "location_type": self.job.floor_loctype.name,
+                },
+                attrs={
+                    "tenant": self.job.dnac.location.tenant.name if self.job.dnac.location.tenant else None,
+                    "uuid": None,
+                },
+            )
+        if (
+            self.job.dnac.location.parent.parent
+            and self.job.dnac.location.parent.parent.location_type == self.job.building_loctype
+        ):
             self.get_or_instantiate(
                 self.building,
-                ids={"name": self.job.dnac.location.name},
+                ids={
+                    "name": self.job.dnac.location.parent.parent.name,
+                    "location_type": self.job.area_loctype.name,
+                    "parent": (
+                        self.job.dnac.location.parent.parent.parent.name
+                        if self.job.dnac.location.parent.parent.parent
+                        else None
+                    ),
+                },
```

Comment:
I'm not familiar with how dnac works so this might be a dumb question. Do you need to account for `self.job.dnac.location.parent = None`?

Reply:
I believe there's a check for parent being required for the LocationType or not.

Reply:
I've added a function to validate the LocationType's that are specified are related and configured correctly. However, regarding your question, a building MUST exist within an Area so there should always be a parent for them.